java/src/main/java/ai/onnxruntime/OrtAllocator.java

```diff
@@ -4,45 +4,32 @@
  */
 package ai.onnxruntime;
 
-import java.io.IOException;
-
 /** An ONNX Runtime memory allocator. */
-class OrtAllocator implements AutoCloseable {
+class OrtAllocator extends NativeObject {
+
+  static final OrtAllocator DEFAULT_ALLOCATOR;
 
   static {
     try {
-      OnnxRuntime.init();
-    } catch (IOException e) {
-      throw new RuntimeException("Failed to load onnx-runtime library", e);
+      DEFAULT_ALLOCATOR = new OrtAllocator(getDefaultAllocator(OnnxRuntime.ortApiHandle), true);
```

Comment:
It wasn't clear to me if the default allocator could exist separately from the OrtEnvironment. It looks like this allocator has a different lifespan, have you found any issues from that?

Reply:
I checked the C API example to be sure: https://github.com/microsoft/onnxruntime/blob/master/csharp/test/Microsoft.ML.OnnxRuntime.EndToEndTests.Capi/C_Api_Sample.cpp#L64

The allocator seems separate from the environment, so it appears to be safe to move this.

Reply:
Ok. We should get someone from MS to check this. I think you're right, it might have been the initial version of this that I wrote against the 0.5.0 API which had allocator issues.

Reply:
Default allocator is a singleton. There is a PR in works that is making things more evident in C#.

---
In reply to: [445867761](https://github.com/microsoft/onnxruntime/pull/4329#discussion_r445867761) [](ancestors = 445867761)